src/SAML2/XML/ExtendableElementTrait.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace SAML2\XML;
+
+use DOMElement;
+use SAML2\Utils;
+use SAML2\XML\alg\AbstractAlgElement as ALG;
+use SAML2\XML\alg\DigestMethod;
+use SAML2\XML\alg\SigningMethod;
+use SAML2\XML\mdattr\EntityAttributes;
+use SAML2\XML\mdrpi\AbstractMdrpiElement as MDRPI;
+use SAML2\XML\mdrpi\PublicationInfo;
+use SAML2\XML\mdrpi\RegistrationInfo;
+use SAML2\XML\mdui\Common as MDUI;
+use SAML2\XML\mdui\DiscoHints;
+use SAML2\XML\mdui\UIInfo;
+use SAML2\XML\shibmd\Scope;
+
+/**
+ * Trait for metadata elements that can be extended.
+ *
+ * @package simplesamlphp/saml2
+ */
+trait ExtendableElementTrait
```

Comment:
I changed this one yesterday. Basically, the entire way extensions were implemented was overcomplicated and nonsensical. I tried to simplify everything by having `md:Extensions` do all the heavy lifting in the two standard `fromXML()` and `toXML()` methods (it now extends `AbstractXMLElement`), and then this trait just adds the property with the `SAML2\XML\md\Extensions` object and the common getter and setter methods.

Reply:
I have indeed missed that due to the filename/classname change.. Fixed

Reply:
Something's off in this file with the Extensions-property.. phpdoc says it's an array of extensions, but we're not treating it that way.. What did you intent to do?